Encrypt the rust DB by default, to match EX

kegsay opened the pull request.

kegsay committed:
Encrypt the rust DB by default, to match EX

andybalaam reviewed (approved):
Looks sensible if/when you can get the tests to pass

kegsay commented:
Blocked until performance on rust SDK improves, as it's spiking CI runtimes too much (3-5x).

richvdh commented:
Looks like we could have another go at this, now that SqliteStore supports `open_with_key` (though maybe we need to update the ClientBuilder too?)

poljar commented:
> Looks like we could have another go at this, now that SqliteStore supports `open_with_key` (though maybe we need to update the ClientBuilder too?)

The builder in the bindings is getting updated in https://github.com/matrix-org/matrix-rust-sdk/pull/6581.

richvdh committed:
Merge remote-tracking branch 'origin/main' into kegan/encrypt-rust-db

richvdh committed:
Update to modern API

The `Passphrase` method got renamed to `SessionPassphrase`, and then removed
altogether in
https://github.com/matrix-org/matrix-rust-sdk/commit/7c6ff517d5bdd60f3f7f547bc3a7e016fc72dc51.

The modern way is to make our own SqliteStoreBuilder, instead of having
SessionPaths do it for us.
